app/models/queue_item.rb

```diff
@@ -5,13 +5,26 @@ class QueueItem < ActiveRecord::Base
   delegate :category, to: :video
   delegate :title, to: :video, prefix: :video
   
+  validates_numericality_of :position, { only_integer: true }
   def rating
-    review = Review.where(user_id: user.id, video_id: video.id).first
     review.rating if review
   end
   
+  def rating=(new_rating)
+    if review
+      review.update_column(:rating, new_rating)
+    else
+      review = Review.new(user: user, video: video, rating: new_rating)
+      review.save(validate: false)
+    end
+  end
+  
   def category_name
     category.name
   end
-
+  
+  private
+    def review
+      @review ||= review = Review.where(user_id: user.id, video_id: video.id).first
```

Comment:
Searching for a single record - you should be using `.find_by` instead of `where(args).first`. Kevin might show the `where` version, but its because videos were recording when Rails 3 was out and in that time it did not matter. However in Rails 4 `.first` sorts the collection prior taking first record so it does an extra step compared to `find_by`. 

Nowadays - the `find_by` is preferred method when you search for 1 record and `where` for collection.